Add noarch python, pip check, entry_point

bollwyvl opened the pull request:
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Hi folks, thanks for maintaining! 

Looks like this is causing the failed builds of `spacy 3` on non-x86. Turns out if the only platform/python-specific dep is `dataclasses`, it's got a no-op variant for py37+ so you it can safely be depended upon without any side effects, and do many, many fewer builds. 

Also, while i was at it:
- add `pip check`
  - those pins look... aggressive. better to be sure!
- add missing CLI `entry_point`
  - you never know what you'll get without it... _especially_ with `noarch:python`!

Thanks again!

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

bollwyvl commented:
hey hey! `pip check` already worth its weight in :medal_sports: !

bollwyvl commented:
More adventures!

I've removed the (outdated) patch, and now grabbing the requirements.txt from the upstream repo. Also added a CLI check of that entry_point... if it comes up :heavy_check_mark: no further changes planned.

adrianeboyd commented:
Thanks for taking a look at this! That behavior for `dataclasses` is ... on the undocumented side, but if it works, then that would indeed make this package a lot simpler.

I think the `smart_open` upper pin in the original recipe was a typo on my end (also starting at build number 2, sigh), and I'd rather keep the requirements in sync with the upstream package.
